### ISSUE TITLE
Why is there a rehydrate for every route?

### ISSUE THREAD

**Issue by ghost** [redux-offline/redux-offline#190]
Hi, I have a couple of questions about rehydration:

-  I do a GET request in componentWillMount that happens as soon as the user logs in, and that I want to offline if fails. The request happens before rehydration. Rehydration then takes place and, if the connection is down and the GET fails, removes the GET request from the offline outbox!!

- To work around this, I 'gate', as mentioned in #149 , and a couple of other places. This causes a delay, which is understandable if rehydration happens once, after logging in. What I don't understand is why there's rehydration for every route, and is it possible to not rehydrate once the user is logged in?

Bottom line: what is the best way to not lose that GET request that happens before hydrate and yet not get the delay for every page the user goes to. Some pages don't even need all the data, or any data at all...

Thanks!

**Comment by sorodrigo** [comment]
How are you navigating inside you application? Are you using a frontend router? If your doing a SPA the rehydrate should only be called once, that is when you land in the app.

**Comment by ghost** [comment]
Yes, I am using routing and indeed, the rehydrate doesn't happen when I click a route to go to another page.
The rehydrates happens only when the user refreshes the page or enters a new path in the browser navigation bar. 
I take it that this is the expected behavior and there's no way to avoid a rehydrate for these scenarios...
Thank you for the prompt response sorodrigo.

**Comment by sorodrigo** [comment]
I wonder if maybe you could avoid setting up the store on the pages that don't need the data via code-splitting.

**Comment by ghost** [comment]
Good idea. Will do. 
Thanks Rodrigo.
